daemon/algod/api/server/v2/dryrun.go

```diff
@@ -541,8 +541,11 @@ func doDryrunRequest(dr *DryrunRequest, response *generated.DryrunResponse) {
 						err = fmt.Errorf("cost budget exceeded: budget is %d but program cost was %d", allowedBudget-cumulativeCost, cost)
 					}
 				}
-				cost64 := uint64(cost)
-				result.Cost = &cost64
+				// amount the budget was increased
+				budgetDebit := uint64(proto.MaxAppProgramCost * numInnerTxns(delta))
+				budgetCredit := uint64(cost) + budgetDebit
```

Comment:
Why isn't `budgetCredted` just `cost`? It is because `cost` is really `credit - debit` at this point? This might warrant a comment.